pkg/reconciler/pipeline/controller.go

```diff
@@ -725,6 +726,20 @@ func (r *pipelineReconciler) updateDesiredState(ctx context.Context, pl *dfv1.Pi
 }
 
 func (r *pipelineReconciler) resumePipeline(ctx context.Context, pl *dfv1.Pipeline) (bool, error) {
+
+	// reset pause timestamp
+	if pl.GetAnnotations()[dfv1.KeyPauseTimestamp] != "" {
+		pl.GetAnnotations()[dfv1.KeyPauseTimestamp] = ""
+		body, err := json.Marshal(pl)
+		if err != nil {
+			return false, err
+		}
+		err = r.client.Patch(ctx, pl, client.RawPatch(types.MergePatchType, body))
+		if err != nil && !apierrors.IsNotFound(err) {
+			return false, err
```

Comment:
In this case, we should let it re-queue, right?

Reply:
For this case, yes. We should requeue the pipeline if there is a patching error that is not IsNotFound
```
		if err != nil {
			// if pipeline cannot be found, do not requeue
			if apierrors.IsNotFound(err) {
				return false, nil
		        // otherwise, requeue pipeline
			} else {
				return true, err
			}
		}
```